Dark Mode: new custom theme extension syntax (#1174)
strapi/documentation: docs/developer-docs/latest/development/admin-customization.md

| `locales`       | Array of Strings | Defines availables locales (see [updating locales](/developer-docs/latest/development/admin-customization.md#locales)) |
| `translations`  | Object           | [Extends the translations](#extending-translations)                                                                                                                       |
| `menu`          | Object           | Accepts the `logo` key to change the [logo](#logos) in the main navigation                                                           |

> | `theme.light` and `theme.dark` | Object           | Override or [extend the theme](#theme-extension) for Light and Dark modes                                                         |

Reviewer: I'd settle for "overwrite" here. While it is theoretically possible to extend the theme, one of our DS components can't make use of an extension. Therefore overwriting properties is probably the most common case.

Wdyt?

```suggestion
| `theme.light` and `theme.dark` | Object           | [Overwrite theme properties](#theme-extension) for Light and Dark modes                                                         |
```

    },
    // Override or extend the theme
    theme: {

>       light: {
>         colors: {
>           primary100: '#f6ecfc',
>           primary200: '#e0c1f4',
>           primary500: '#ac73e6',
>           primary600: '#9736e8',
>           primary700: '#8312d1',
>           danger700: '#b72b1a'
>         },
>       }

Reviewer: ```suggestion
      // overwrite light theme properties
      light: {
        colors: {
          primary100: '#f6ecfc',
          primary200: '#e0c1f4',
          primary500: '#ac73e6',
          primary600: '#9736e8',
          primary700: '#8312d1',
          danger700: '#b72b1a'
        },
      },
      
      // overwrite dark theme properties
      dark: {
         // ...
      }
```

#### Theme extension

> Strapi applications can be displayed either in Light or Dark mode (see [administrator profile setup in the User Guide](/user-docs/latest/getting-started/introduction.md#setting-up-your-administrator-profile)), and both can be extended through custom theme development.

Reviewer: ```suggestion
Strapi applications can be displayed either in Light or Dark mode (see [administrator profile setup in the User Guide](/user-docs/latest/getting-started/introduction.md#setting-up-your-administrator-profile)), and both can be extended through custom theme settings.
```

To extend the theme, use either:

- the `config.theme.light` key for the Light mode

> - the `config.theme.dark` key for the Dark mode

Reviewer: I'd omit that and add a comment in the above code example instead.

Author: I don't think a comment in the code example should replace this explanation, especially since otherwise there's no proper information about how to extend the theme in the "Theme extension" section. I'll leave this here. But I integrated all you other suggestions, thank you for your review!

::: strapi Strapi Design System
The default [Strapi theme](https://github.com/strapi/design-system/tree/main/packages/strapi-design-system/src/themes) defines various theme-related keys (shadows, colors…) that can be updated through the `config.theme.light` and `config.theme.dark` keys in `./admin/src/app.js`. The [Strapi Design System](https://design-system.strapi.io/) is fully customizable and has a dedicated [StoryBook](https://design-system-git-main-strapijs.vercel.app) documentation.
:::

::: caution

> The former syntax `config.theme` to extend the theme will soon be deprecated by Strapi, and will stop being supported when the next major release is available. We encourage you to update any custom theme using the new syntax that supports custom development for both Light and Dark modes.

Reviewer: ```suggestion
The former syntax for `config.theme` without `light` or `dark` keys is deprecated at will be removed in the next major release. We encourage you to update your custom theme to use the new syntax that supports light and dark modes.
```
